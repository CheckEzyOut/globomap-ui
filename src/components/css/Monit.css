/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.monit {
  padding: 8px 0;
}

.monit .monit-loading {
  text-align: center;
  padding: 10px;
  color: #bbb;
}

.monit table {
  width: 100%;
  border-spacing: 0;
}

.monit table th {
  text-align: left;
  font-weight: normal;
}

.monit table th,
.monit table td {
  font-size: 12px;
  padding: 4px 8px;
  border-bottom: 1px solid rgba(157, 172, 169, 0.35);
}

.monit table td {
  border-bottom: 1px solid #dadada;
}

.monit table tr:last-child th,
.monit table tr:last-child td {
  border-bottom: 0;
}

.monit .trigger-not-found {
  text-align: center;
}

.monit .trigger-0,
.monit .trigger-1 {
  color: white;
  width: 2%;
  font-size: 14px;
}

.monit .trigger-0 {
  background-color: #34a853;
}
.monit .trigger-1 {
  background-color: #ea4335;
}
