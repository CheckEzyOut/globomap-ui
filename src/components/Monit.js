import React, { Component } from 'react';
import { uiSocket } from './App';
import './css/Monit.css';

class Monit extends Component {
  monitItems = ['comp_unit'];

  constructor(props) {
    super(props);
    this.socket = uiSocket();
    this.state = {
      loading: true,
      triggers: []
    }
  }

  render() {
    if(this.props.node.type === 'comp_unit'){
      let props = this.state.triggers.map((trigger, i) => {
        return <tr key={trigger.triggerid}>
                <th>{trigger.description}</th>
                <td className={'trigger-' + trigger.value}>
                  {this.getIcon(trigger.value)}
                </td>
              </tr>;
      });

      if(props.length === 0) {
        props = [<tr key={1}><th className="trigger-not-found">Not found</th></tr>];
      }

      return <div className="monit">
              {!this.state.loading
                ? <table>
                    <tbody>{props}</tbody>
                  </table>
                : <div className="monit-loading">
                    <i className="fa fa-cog fa-spin fa-2x fa-fw"></i>
                  </div>}
            </div>;
    }
    return null;
  }

  getIcon(val) {
    return parseInt(val, 10) !== 0
           ? <i className="fa fa-times"></i>
           : <i className="fa fa-check"></i>;
  }

  componentWillReceiveProps(nextProps){
    let next = nextProps.node,
        current = this.props.node;

    if(this.monitItems.indexOf(next.type) < 0) {
      return;
    }

    if(current._id !== next._id) {
      this.setState({ loading: true, triggers: [] }, () => {
        this.socket.emit('getmonitoring', next, (data) => {
          if (data.error) {
            console.log(data.message);
            return this.setState({ loading: false, triggers: [] });
          }
          return this.setState({ loading: false, triggers: data });
        });
      });
    }
  }

  componentDidMount() {
    let node = this.props.node;

    if(this.monitItems.indexOf(node.type) < 0) {
      return;
    }

    this.setState({ loading: true, triggers: [] }, () => {
      this.socket.emit('getmonitoring', node, (data) => {
        if (data.error) {
          console.log(data.message);
          return this.setState({ loading: false, triggers: [] });
        }
        return this.setState({ loading: false, triggers: data });
      });
    });
  }

}

export default Monit;
