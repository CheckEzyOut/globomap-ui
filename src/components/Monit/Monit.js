/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Loading } from '../';
import { showModal, closeModal } from '../../redux/modules/app';
import { getZabbixMonitoring,
         getZabbixGraph } from '../../redux/modules/plugins';
import './Monit.css';

export class Monit extends Component {
  monitItems = ['comp_unit', 'zabbix_graph'];

  getIcon(val) {
    return parseInt(val, 10) !== 0
            ? <i className="fa fa-times"></i>
            : <i className="fa fa-check"></i>;
  }

  openZbxGraph(event) {
    event.stopPropagation();
    const node = this.props.node;

    if (!node)
      return;

    this.props.showModal(<img src={`data:image/png;base64,${this.props.zbxGraph}`}
                              alt={node.name} />);
  }

  componentWillReceiveProps(nextProps) {
    const next = nextProps.node;
    const current = this.props.node;

    if(!this.monitItems.includes(next.type)) {
      return;
    }

    if(current._id !== next._id) {
      if (current.type === 'comp_unit') {
        this.props.getZabbixMonitoring(next);
      }

      if (current.type === 'zabbix_graph') {
        this.props.getZabbixGraph(next);
      }
    }
  }

  componentDidMount() {
    const node = this.props.node;

    if(!this.monitItems.includes(node.type)) {
      return;
    }

    if (node.type === 'comp_unit') {
      this.props.getZabbixMonitoring(node);
    }

    if (node.type === 'zabbix_graph') {
      this.props.getZabbixGraph(node);
    }
  }

  render() {
    if (!this.props.node)
      return null;

    let node = this.props.node,
        triggers = [],
        zbxGraphButton = '';

    if (node.type === 'comp_unit') {
      triggers = this.props.zbxTriggers.map((trigger) => {
        return (
          <tr key={trigger.properties.triggerid}>
            <th>{trigger.key}</th>
            <td className={'trigger-' + trigger.value}>
              {this.getIcon(trigger.value)}
            </td>
          </tr>
        );
      });

      if (triggers.length === 0) {
        triggers = [<tr key={1}><th className="trigger-not-found"></th></tr>];
      }
    }

    if (node.type === 'zabbix_graph') {
      zbxGraphButton = (
        <button className="topcoat-button--cta" onClick={(e) => this.openZbxGraph(e)}>
          <i className="fa fa-chart-bar"></i>&nbsp;Show Zabbix Graph
        </button>
      );
    }

    return (
      <div className="monit">
        <div className="monit-buttons">
          { zbxGraphButton }
        </div>
        <table>
          <tbody>{ triggers }</tbody>
        </table>
        <Loading iconSize="medium"
                 isLoading={ this.props.zbxMonitLoading ||
                             this.props.zbxGraphLoading } />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    zbxMonitLoading: state.plugins.zbxMonitLoading,
    zbxGraphLoading: state.plugins.zbxGraphLoading,
    zbxTriggers: state.plugins.zbxTriggers,
    zbxGraph: state.plugins.zbxGraph
  };
}

export default connect(
  mapStateToProps,
  { getZabbixMonitoring, getZabbixGraph, showModal,
    closeModal }
)(Monit);
