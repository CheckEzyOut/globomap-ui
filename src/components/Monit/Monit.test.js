import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { Monit } from './Monit';

describe("Monit", function() {

  let componentDidMountStub;

  beforeEach(function() {
    componentDidMountStub = sinon.stub(Monit.prototype, 'componentDidMount').value('newValue');
  });

  afterEach(function () {
    componentDidMountStub.restore();
  });

  it('should render Monit component', () => {
    const component = shallow(<Monit />);
    expect(component).toMatchSnapshot();
  });

});
