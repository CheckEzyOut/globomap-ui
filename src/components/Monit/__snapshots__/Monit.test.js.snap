// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monit should render Monit component 1`] = `""`;
